build/Rulesets/Roslyn_BuildRules.ruleset

```diff
@@ -81,23 +81,23 @@
     <Rule Id="RS0027" Action="Error" />
   </Rules>
   <Rules AnalyzerId="System.Collections.Immutable.Analyzers" RuleNamespace="System.Collections.Immutable.Analyzers">
-    <Rule Id="RS0012" Action="Warning" />
```

Comment:
Here's the prerequisites:
1. Fix the implementation of IHasArgumentsExpression
2. update analyzers to use the new API, verify their behavior is correct, and release them
3. update roslyn to reference updated analyzer

Then we can revert this change if everything runs well. For now, I will create an issue in roslyn repo to track this. FYI @jinujoseph